client_test.go

```diff
@@ -256,6 +407,500 @@ func TestProvisionCredentials(t *testing.T) {
+	})
+}
+
+func TestDeprovisionResource(t *testing.T) {
+	t.Run("202 accepted", func(t *testing.T) {
+		gm.RegisterTestingT(t)
+
+		srv := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			rw.Header().Add("Content-Type", "application/json")
+			rw.WriteHeader(http.StatusAccepted)
+			rw.Write([]byte(`{"message":"please wait"}`))
+		}))
+		defer srv.Close()
+
+		message, async, err := callDeprovisionResource(srv.URL)
+
+		gm.Expect(err).ToNot(gm.HaveOccurred())
+		gm.Expect(async).To(gm.BeTrue(), "Result on 202 should be async")
+		gm.Expect(message).To(gm.Equal("please wait"))
+	})
+
+	t.Run("202 no message", func(t *testing.T) {
+		gm.RegisterTestingT(t)
+
+		srv := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			rw.Header().Add("Content-Type", "application/json")
+			rw.WriteHeader(http.StatusAccepted)
+			rw.Write([]byte(`{}`))
+		}))
+		defer srv.Close()
+
+		msg, async, err := callDeprovisionResource(srv.URL)
+
+		gm.Expect(err).ToNot(gm.HaveOccurred())
+		gm.Expect(async).To(gm.BeTrue(), "Result on 202 should be async")
+		gm.Expect(msg).To(gm.Equal(""))
+	})
+
+	t.Run("400 bad request valid response", func(t *testing.T) {
+		gm.RegisterTestingT(t)
+		srv := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			rw.Header().Add("Content-Type", "application/json")
+			rw.WriteHeader(http.StatusBadRequest)
+			rw.Write([]byte(`{"message":"i dont get ya"}`))
+		}))
+		defer srv.Close()
+
+		_, _, err := callDeprovisionResource(srv.URL)
+
+		gm.Expect(err).To(gm.MatchError(NewError(errors.BadRequestError, "i dont get ya")))
+	})
+
+	t.Run("400 bad request with invalid content-type", func(t *testing.T) {
+		gm.RegisterTestingT(t)
+		srv := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			rw.Header().Add("Content-Type", "text/html")
+			rw.WriteHeader(http.StatusBadRequest)
+			rw.Write([]byte(`{"message":"i dont get ya"}`))
+		}))
+		defer srv.Close()
+
+		_, _, err := callDeprovisionResource(srv.URL)
+
+		gm.Expect(err).To(gm.MatchError(`no consumer: "text/html"`))
+	})
+
+	t.Run("400 bad request with missing message", func(t *testing.T) {
+		gm.RegisterTestingT(t)
+		srv := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			rw.Header().Add("Content-Type", "application/json")
+			rw.WriteHeader(http.StatusBadRequest)
+			rw.Write([]byte(`{}`))
+		}))
+		defer srv.Close()
+
+		_, _, err := callDeprovisionResource(srv.URL)
+
+		gm.Expect(err).To(gm.MatchError(ErrMissingMsg))
+	})
+
+	t.Run("401 unauthorized valid response", func(t *testing.T) {
+		gm.RegisterTestingT(t)
+		srv := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			rw.Header().Add("Content-Type", "application/json")
+			rw.WriteHeader(http.StatusUnauthorized)
+			rw.Write([]byte(`{"message":"i dont get ya"}`))
+		}))
+		defer srv.Close()
```

Comment:
most of the tests using a server only vary the response in their status code, so you could do something like:

```go
t.Run("401 unauthorized valid response", withCode(http.StatusUnauthorized, func(url string) {
        _, _, err := callDeprovisionResource(srv.URL)
  	gm.Expect(err).To(gm.MatchError(NewError(errors.UnauthorizedError, "i dont get ya")))
})

/* ... */

func withCode(code http.StatusCode, fn func(string)) func(t *testing.T) {
        return func(t *testing.T) {
                gm.RegisterTestingT(t)
		srv := httptest.NewServer(http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
 			rw.Header().Add("Content-Type", "application/json")
 			rw.WriteHeader(code)
 			rw.Write([]byte(`{"message":"i dont get ya"}`))
 		}))
 		defer srv.Close()
                fn(srv.URL)
        }
}